core/src/main/java/com/linecorp/armeria/common/RequestContextExporterBuilder.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.util.Objects.requireNonNull;
+
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.RequestContextExporter.ExportEntry;
+
+import io.netty.util.AsciiString;
+import io.netty.util.AttributeKey;
+
+/**
+ * Builds a new {@link RequestContextExporter}.
+ */
+public final class RequestContextExporterBuilder {
+
+    private static final String PREFIX_ATTRS = "attrs.";
+    private static final String PREFIX_HTTP_REQ_HEADERS = "req.http_headers.";
+    private static final String PREFIX_HTTP_RES_HEADERS = "res.http_headers.";
+
+    private final Set<BuiltInProperty> builtIns = EnumSet.noneOf(BuiltInProperty.class);
+    private final Set<ExportEntry<AttributeKey<?>>> attrs = new HashSet<>();
+    private final Set<ExportEntry<AsciiString>> httpReqHeaders = new HashSet<>();
+    private final Set<ExportEntry<AsciiString>> httpResHeaders = new HashSet<>();
+
```

Comment:
Missing package-private default constructor. This class must be instantiated only via `RequestContextExporter.builder()`.

Reply:
Oops! Thanks for detecting it.🙏